From a1842a00589b5fd67a9e83b16ebe80973f910616 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:39:57 -0400
Subject: [virt] kvm/vmx: DR7 masking on task switch emulation is wrong

Message-id: <1406040016-3289-203-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86281
O-Subject: [RHEL7 PATCH v2 202/221] KVM: vmx: DR7 masking on task switch emulation is wrong
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Nadav Amit <namit@cs.technion.ac.il>

The DR7 masking which is done on task switch emulation should be in hex format
(clearing the local breakpoints enable bits 0,2,4 and 6).

Signed-off-by: Nadav Amit <namit@cs.technion.ac.il>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit 1f854112553a1d65363ab27d4ee3dfb4b27075fb)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index 6f7463f..f30d00f 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -5416,7 +5416,7 @@ static int handle_task_switch(struct kvm_vcpu *vcpu)
  }
 
  /* clear all local breakpoint enable flags */
- vmcs_writel(GUEST_DR7, vmcs_readl(GUEST_DR7) & ~55);
+ vmcs_writel(GUEST_DR7, vmcs_readl(GUEST_DR7) & ~0x55);
 
  /*
   * TODO: What about debug traps on tss switch?
-- 
1.7.1